lib/recording/RecordingManager.js

```diff
@@ -43,119 +44,136 @@ class RecordingManager {
                     })
             };
 
             let startRecordingTime = new Date();
-            logger.debug("[%s-%s] Start recording ", entryObj.entryId, entryObj.recordedEntryId);
-            let recordingEntrySession = getRecordingEntrySession();
-            if (recordingEntrySession == null) {
-                return Q.resolve();
-            }
-            let recordingSessionPath = recordingEntrySession.recordingSessionPath;
-            let startSessionPromise;
-            return qio.exists(recordingSessionPath).then((sessionExist)=> {
-                if (sessionExist) {
-                    startSessionPromise = startExistSession()
-                }
-                else {
-                    startSessionPromise = startNewSession()
-                }
-                return startSessionPromise
-                    .then(()=> {
-                        let startRecordingTimeDuration = new Date() - startRecordingTime;
-                        logger.info("[%s-%s] Start recording is completed, time : %s ms", entryObj.entryId, entryObj.recordedEntryId, startRecordingTimeDuration)
-                    })
-            })
+            logger.debug(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Start recording`);
+            return getRecordingEntrySession.call(this, entryObj)
+                .then((recordingEntrySession)=> {
+                    if (recordingEntrySession == null) {
+                        return Q.resolve();
+                    }
+                    let startSessionPromise;
+                    return qio.exists(recordingEntrySession.recordingSessionPath).then((sessionExist)=> {
+                        if (sessionExist) {
+                            startSessionPromise = startExistSession(recordingEntrySession);
+                        }
+                        else {
+                            startSessionPromise = startNewSession(recordingEntrySession);
+                        }
+                        return startSessionPromise
+                            .then(()=> {
+                                let startRecordingTimeDuration = new Date() - startRecordingTime;
+                                logger.info(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Start recording is completed, time : ${startRecordingTimeDuration} ms`);
+                            })
+                    });
+                })
+                .catch((err)=> {
+                    // In case of error, pass message on and print to log
+                    return Q.reject(err);
+                });
         }).catch((err)=>{
-            logger.error("[%s-%s] Error while try to start recording : %s, delete object", entryObj.entryId, entryObj.recordedEntryId, ErrorUtils.error2string(err))
+            logger.error(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Error while try to start recording : ${ErrorUtils.error2string(err)}, delete object`);
             delete this.recordingList[entryObj.entryId]
```

Comment:
If property doesn't exist in object delete has no effect and returns, so don't see the need to add another checkup here.
You can see the following: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete